### PR TITLE
feat: add public accessors to error class

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Adds public accessors to Error, for use by 3rd party developers and internal API, and nullability annotations.

Note: the intention is for further work to be carried out in a separate PR (extracting the data capture to a separate class, etc)

### Changed

Converted fields/methods to properties (getter/setter backed with a field) in Error.

## Tests

Adapted unit tests for the default value + override value of each accessor.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: sdk/src/main/java/com/bugsnag/android/Error.java [hunk 1]
```diff
@@ -112,7 +134,7 @@ public void setContext(String context) {
      */
     @Nullable
     public String getContext() {
-        if (!TextUtils.isEmpty(context)) {
+        if (context != null) {
```
Comment (martin308) [inline sdk/src/main/java/com/bugsnag/android/Error.java]: what's the intent of this change?

Comment (fractalwrench) [reply sdk/src/main/java/com/bugsnag/android/Error.java]: This was unintentional - I've reverted back to the TextUtils.isEmpty approach which is more robust.
